st-embroider/2024/2024-08-06.md

```diff
@@ -0,0 +1,49 @@
+# 2024-08-06
+
+Note Taker: Chris Manson
+Time Keeper: Alex
+
+## Attendees
+
+Add yourself to the list if you attend and check the box!
+
+- [x] Ed Falkner (EF)
+- [x] Chris Manson (CM)
+- [ ] Aaron Chambers (AC)
+- [x] Alex (A)
+- [ ] Preston Sego (PS)
+- [x] Simon Ihmig (SI)
+- [x] Katie Gengler (KG)
+- [ ] Peter Meehan (PM)
+- [ ] Krystan HuffMenne (KH)
+- [x] Peter Wagenet (PW)
+- [ ] James Davis (JD)
+- [ ] Marine Dunstetter (MD)
+
+## Topics
+
+### Embroider Roadmap discussion
+
+- Almost there on the vite stuff. Two more steps
+  - stop declaring app as auto-upgraded
+  - stop flattening the app and test folders
+- discussion about v2 addon RFC - simple app booting
+
+### template compiler issue in ember-source https://github.com/emberjs/ember.js/issues/20718 <@mansona>
+
+- EF: the way that it used to work needs to continue to work
+- KG: should this block the release patch? 
+- EF: no go ahead
+- the patch is very important
+- we could debug the issue on the call if we have time
+
+### Glint generic components <@ef4>
+
+- EF: it's not possible to make components that are generic over their components
+- PW: what's the issue? it's just the template only component that has the issue
+- EF: we could implement some way to use class syntax to define a "template only component" only for the types
+- the issue is that we only have 2 base classes `@ember/component` and `@glimmer/component` which both have behaviours
+- we could implement some other base class that gives us the ability to define generics
```

Comment:
@ef4 @wagenet I might be misunderstanding the issue at hand and what you mean by "generic over their components", but isn't `ComponentLike` exactly the type you're describing here?

Reply:
Can't do `type X<Y> = ComponentLike<{ Args: { foo: Y }}>`

Reply:
Why not?

Reply:
you're right, I didn't put enough effort in to my comment.

more fully:
```ts
export const MyComponent: ComponentLike<{ /* no generics */ }> = <template></template>
```

and 
```ts
type Extended<X> = ComponentLike<{ Args: { foo: X }}>;
export const MyComponent: Extended< /* no generics */ > = <template></template>
```

However:
```glimmer-ts
import type { ComponentLike } from '@glint/template';

type Generic<X = unknown> = ComponentLike<{
  Args: {
    x: X;
  };
  Blocks: {
    default: [X];
  };
}>;

export const Demo: Generic = <template>{{@x}}</template>;

export const NoGenerics = <template>
  <Demo @x={{2}} as |num|>
    {{num}}
  </Demo>
  <Demo @x="hello" as |str|>
    {{str}}
  </Demo>
</template>;
```
![image](https://github.com/user-attachments/assets/15fd3acb-de71-4cbb-af1a-39c8db94ca6f)
```
Only primitive values and certain DOM objects (see `ContentValue` in `@glint/template`) are usable as top-level template content.
     Argument of type 'unknown' is not assignable to parameter of type 'ContentValue'. [2345]
```

Reply:
Right, the syntactic issue with template-only components having no way to specify type parameters is understood, and if that's what was being discussed here then I have nothing to add 🙂 

This sounded slightly different, though:

> EF: it's not possible to make components that are generic over their components

I read this as a question of describing a the type of component (or a function or whatever) that accepts a component as an arg but doesn't care whether it's an `@ember/component` or `@glimmer/component`, which was why I was asking for clarity.

Since it sounds like that's not the case, I'll see myself out

Reply:
Here is a related issue around generics completely breaking Glint (even with classes):

https://discord.com/channels/480462759797063690/717767358743183412/1274108269568458834